### PR TITLE
fix(#1587): pure history rows range window iterator initialize position wrongly

### LINKED ISSUE #1587: `at/lag/first_value` udf function over merged window report different result compared to those not merged
**Bug Description**

two compared case: 
1.  case one
```yaml
  - id: 6
    desc: window merge optimization
    inputs:
      - columns: [ "row_id int","ts timestamp","group1 string","val1 int" ]
        indexs: [ "index1:group1:ts" ]
        name: t1
        data: |
          1, 1612130400000, g1, 1
          2, 1612130401000, g1, 2
          3, 1612130402000, g1, 3
          4, 1612130403000, g1, 4
          5, 1612130404000, g1, 5
    sql: |
      select
      `row_id`,
      `val1`,
      lag(`val1`, 0) over w1 as agg1,
      lag(`val1`, 0) over w2 as agg2
      from `t1` WINDOW
        w1 as (partition by `group1` order by `ts` rows_range between 5s preceding and 0s preceding MAXSIZE 10),
        w2 as (partition by `group1` order by `ts` rows_range between 5s preceding and 1s preceding MAXSIZE 10);
```

2.   case two
 ```yaml
  - id: 6
    desc: window merge optimization
    inputs:
      - columns: [ "row_id int","ts timestamp","group1 string","val1 int" ]
        indexs: [ "index1:group1:ts" ]
        name: t1
        data: |
          1, 1612130400000, g1, 1
          2, 1612130401000, g1, 2
          3, 1612130402000, g1, 3
          4, 1612130403000, g1, 4
          5, 1612130404000, g1, 5
    sql: |
      select
      `row_id`,
      `val1`,
      lag(`val1`, 0) over w2 as agg2
      from `t1` WINDOW
        w2 as (partition by `group1` order by `ts` rows_range between 5s preceding and 1s preceding MAXSIZE 10);
```

in case two , `lag(val1, 0) over w1` will produce series `NULL, 1, 2, 3, 4`, while in case one, it produce `NULL, NULL , NULL, NULL, NULL` instead. 

The case do not concern correctness of `lag` function but the consistency of two similar windows before and after merged in compilation, for lag correctness refer #1554 

**Expected Behavior**

**Steps to Reproduce**

1. 
2. 
4. 
5.

### PR DESCRIPTION
**Pull request by aceforeverd**
**TL;DR**

InnerRangeList should call `SeekToFirst` on construct, close #1587 

previous code use `root_->SeekToFirst`, which result `root_` to the
current row instead the first element inside window. That is different
when the window is a pure history window

### CODE REVIEW

**Comment by aceforeverd** [comment]
[High-priority Bugs](https://trello.com/c/h7opm5kE/45-high-priority-bugs)

**Review by zhanghaohit — APPROVED** [review]
LGTM

**Review by vagetablechicken — APPROVED** [review]
